lib/graph_records.rb

```diff
@@ -27,41 +210,71 @@ def graph(**objects)
     @inspected = Set.new
     @focus = @focus_config.map { _1.to_s.classify.constantize.where(id: _2) }
 
-    objects.map do |klass, ids|
-      class_name = klass.to_s.singularize
-      associated_objects =
-        load_association(
-          class_name,
-          class_name.classify.constantize.where(id: ids)
+    if @primary_type.nil?
+      @primary_type =
+        (
+          if objects.keys.size >= 1
+            objects.keys.first.to_s.singularize.to_sym
+          else
+            :patient
+          end
         )
```

Comment:
Ruby has a handy operator `||=` that allows you to assign a variable if it's not already assigned a value.

```suggestion
    @primary_type ||=
      if objects.keys.size >= 1
        objects.keys.first.to_s.singularize.to_sym
      else
        :patient
      end
```